Src/tim.c

```diff
@@ -53,9 +53,9 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Init.Prescaler = 23; // A prescaler of 23 brings the frequency from 48 MHz to 2 MHz (since 48/(23+1) = 2
```

Comment:
you can get slightly better resolution by choosing a better prescler, but this is good enough!